h2o-py/tests/testdir_algos/gam/pyunit_generate_synthetic_GAM_data.py

```diff
@@ -0,0 +1,95 @@
+import sys
+sys.path.insert(1,"../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator as gam
+
+# This test will generate synthetic GAM dataset.  If given to a GAM model, it should be able to perform well with 
+# this dataset since the assumptions associated with GAM are used to generate the dataset.
+def test_define_dataset():
+    family = 'binomial' # can be any valid GLM families
+    nrow = 100000
+    ncol = 10
+    realFrac = 0.4
+    intFrac = 0.3
+    enumFrac = 0.3
+    missing_fraction = 0
+    factorRange= 50
+    numericRange = 10
+    targetFactor = 1
+    numGamCols = 1
+    
+    assert numGamCols <= ncol*realFrac, "Number of real columns {0} should exceed the number of gam columns " \
+                                       "{1}".format(ncol*realFrac, numGamCols) # gam can be only real columns
+    gamDataSet = generate_dataset(family, nrow, ncol, realFrac, intFrac, enumFrac, missing_fraction, factorRange, 
+                                  numericRange, targetFactor, numGamCols)
+    #h2o.download_csv(gamDataSet, "/Users/.../dataset.csv") # save dataset
+    assert gamDataSet.nrow == nrow, "Dataset number of row: {0}, expected number of row: {1}".format(gamDataSet.nrow, 
+                                                                                                     nrow)
+    assert gamDataSet.ncol == (1+ncol), "Dataset number of row: {0}, expected number of row: " \
+                                                          "{1}".format(gamDataSet.ncol, (1+ncol))
```

Comment:
since the goal is to create a good dataset for GAM, we should perhaps run GAM on it an check we get a very good accuracy

Reply:
Will do before I merge